### PR TITLE
Perform a type check upon deserialization

### PR DESCRIPTION
**Pull request by ChristophWurst**
Otherwise any garbage data will cause a type error due to the mismatching return value and the return type hint.

### CODE REVIEW

File: src/Persisters/Serializers/Native.php [hunk 1]
```diff
@@ -3,6 +3,9 @@
 namespace Rubix\ML\Persisters\Serializers;
 
 use Rubix\ML\Persistable;
+use RuntimeException;
+use function get_class;
+use function is_object;
```
Comment (andrewdalpino) [inline src/Persisters/Serializers/Native.php:9]: Classy function imports, nice touch, we generally put a space between the last class import and the first function import (same with imported constants). Check the Travis logs for the messages coming from cs-fixer.

`$ composer fix` will automatically format the code

https://travis-ci.org/github/RubixML/RubixML/builds/691331649

File: tests/Persisters/Serializers/NativeTest.php [hunk 2]
```diff
@@ -42,19 +45,41 @@ public function build() : void
...
+        $this->expectException(RuntimeException::class);
+
+        $this->serializer->unserialize($data);
+    }
+
+    /**
+     * @test
+     */
+    public function deserializeNonPersistable() : void
+    {
```
Comment (andrewdalpino) [inline tests/Persisters/Serializers/NativeTest.php:84]: Is this 'essentially' a duplicate test as the one above? Could it be factored into one test or perhaps a data provider would be helpful (in which case the method name could be something like deserializeBadData?

File: tests/Persisters/Serializers/NativeTest.php [hunk 3]
```diff
@@ -7,6 +7,9 @@
 use Rubix\ML\Persisters\Serializers\Native;
 use Rubix\ML\Persisters\Serializers\Serializer;
 use PHPUnit\Framework\TestCase;
+use RuntimeException;
+use stdClass;
```
Comment (andrewdalpino) [inline tests/Persisters/Serializers/NativeTest.php:11]: Space here

Run `$ composer check` to run the style checker, or `$composer fix` to automatically format the code per our style guide.

**Review by andrewdalpino — CHANGES_REQUESTED** [review]
Nice catch @ChristophWurst

Had a question about the tests and Travis had a few things to say as well but other than that it looks great. Thanks for the PR welcome to Rubix

File: tests/Persisters/Serializers/NativeTest.php (hunk 2 above)
Comment (ChristophWurst) [reply tests/Persisters/Serializers/NativeTest.php:84]: I copy-pasted and forgot to change `stdClass` to something that is not an object. In any case, combining the two similar methods makes sense.

File: tests/Persisters/Serializers/NativeTest.php [hunk 4]
```diff
@@ -42,19 +46,45 @@ public function build() : void
...
 
         $persistable = $this->serializer->unserialize($data);
 
         $this->assertInstanceOf(DummyClassifier::class, $persistable);
         $this->assertInstanceOf(Persistable::class, $persistable);
     }
+
+	/**
+	 * @return array<mixed>
+	 */
```
Comment (andrewdalpino) [inline tests/Persisters/Serializers/NativeTest.php]: lol you made the change before i could comment ... this works, but should the annotation technically be `array[]`?

btw `$ composer analyze` to run static analysis

File: tests/Persisters/Serializers/NativeTest.php [hunk 5]
```diff
@@ -42,19 +46,45 @@ public function build() : void
...
         $this->assertIsString($data);
 
         $persistable = $this->serializer->unserialize($data);
 
         $this->assertInstanceOf(DummyClassifier::class, $persistable);
         $this->assertInstanceOf(Persistable::class, $persistable);
     }
+
+    /**
+     * @return array<mixed>
```
Comment (andrewdalpino) [inline tests/Persisters/Serializers/NativeTest.php:66]: lol made changes before I could drop the last comment ... this works, `array[]` perhaps could be more specific ... not a big deal

**Review by andrewdalpino — APPROVED** [review]

File: tests/Persisters/Serializers/NativeTest.php (hunk 5 above)
Comment (ChristophWurst) [reply tests/Persisters/Serializers/NativeTest.php:66]: phpstan inspired me to this :shrug: :speak_no_evil:

**Comment by andrewdalpino** [comment]
Thanks again @ChristophWurst 

Feel free to join our chat room on Telegram https://t.me/RubixML

**Comment by ChristophWurst** [comment]
Cheers!

I don't have Telegram, sorry :)

File: tests/Persisters/Serializers/NativeTest.php (hunk 5 above)
Comment (andrewdalpino) [reply tests/Persisters/Serializers/NativeTest.php:66]: I love Stan. Check our contributing guide if you haven't already ... we've got all the custom composer commands listed, we try to use static analysis as much as possible.

https://github.com/RubixML/RubixML/blob/master/CONTRIBUTING.md

**Pull request merged**